File: medium/0451_sort_characters_by_frequency/solution.cpp
#include <string>
#include <unordered_map>

class Solution {
public:
  std::string frequencySort(std::string s) {
    std::unordered_map<char, int> frequencyMap;
    for (const char &ch : s)
      ++frequencyMap[ch];

    std::sort(s.begin(), s.end(), [&](const char &a, const char &b) {
      if (frequencyMap[a] == frequencyMap[b])
        return a > b;

      return frequencyMap[a] > frequencyMap[b];
    });

    return s;
  }
};

File: medium/0451_sort_characters_by_frequency/solution.py
from collections import defaultdict


class Solution:
    def frequencySort(self, s: str) -> str:
        frequencyMap = defaultdict(int)
        for c in s:
            frequencyMap[c] += 1

        return "".join(
            c * frequency
            for c, frequency in sorted(
                frequencyMap.items(), key=lambda item: -item[1]
            )
        )
